model/riscv_sys_regs.sail

```diff
@@ -721,7 +726,266 @@ function legalize_satp32(a : Architecture, o : bits(32), v : bits(32)) -> bits(3
+//   Action : 0 .. 5,
+// }
+
+// bitfield ETrigger : xlenbits = {
+//   // Trigger TYPE bits.
+//   TYPE   : xlen - 4 .. xlen - 1,
+//   // Debug Mode bit
+//   Dmode  : xlen - 5,
+//   // Trigger firing status bit.
+//   Hit    : xlen - 6,
+//   // Reserved WPRI bits.
+//   wpri_2 : 13 .. xlen - 7,  
+//   // Trigger in VS-mode.
+//   VS     : 12,  
+//   // Trigger in VU-mode.
+//   VU     : 11,  
+//   // Reserved WPRI bits.
+//   wpri_1 : 10,
+//   // Trigger in M-mode.
+//   M      : 9,
+//   // Reserved WPRI bits.
+//   wpri_0 : 8,
+//   // Trigger in S-mode.
+//   S      : 7,  
+//   // Trigger in U-mode.
+//   U      : 6 .. 21,      
+//   // Action on Trigger bits.
+//   Action : 0 .. 5,
+// }
+
+// bitfield TMexttrigger : xlenbits = {
+//   // Trigger TYPE bits.
+//   TYPE   : xlen - 4 .. xlen - 1,
+//   // Debug Mode bit
+//   Dmode  : xlen - 5,
+//   // Trigger firing status bit.
+//   Hit    : xlen - 6,
+//   // Reserved WPRI bits.
+//   wpri   : 23 .. xlen - 7,  
+//   // Interrupt control signal bit.
+//   Intctl : 22,    
+//   // Determines the contents of the XLEN-bit compare values.
+//   Select : 6 .. 21,       
+//   // Action on Trigger bits.
+//   Action : 0 .. 5,
+// }
+
+bitfield TControl : xlenbits = {
+  // Reserved WPRI bits.
+  wpri_2 : xlen - 1 .. 8,
+  // M-mode previous trigger enable bit
+  MPTE   : 7,
+  // Reserved WPRI bits.
+  wpri_1 : 6 .. 4,      
+  // M-mode trigger enable bit.
+  MTE    : 3,       
+  // Reserved WPRI bits.
+  wpri_0 : 2 .. 0,
+}
+
+/* Debug/Trace trigger register select */
+register tselect  : TSelect
+/* First Debug/Trace trigger data register */
+register tdata1   : TData1
+/* Second Debug/Trace trigger data register */
+register tdata2   : xlenbits
+/* Third Debug/Trace trigger data register */
+register tdata3   : xlenbits
+/* Match Control Debug/Trace trigger register */
+register mcontrol : MControl
+/* Match Control TYPE 6 Debug/Trace trigger register */
+register mcontrol6: MControl6
+/* Instruction Count Debug/Trace trigger register */
+// register icount   : ICount
+// /* Interrupt Trigger Debug/Trace trigger register */
+// register itrigger : ITrigger
+// /* Exception Trigger Debug/Trace trigger register */
+// register etrigger : ETrigger
+// /* External Trigger Debug/Trace trigger register */
+// register tmexttrigger : TMexttrigger
+/* Debug/Trace trigger Control register */
+register tcontrol : TControl
+/*  Debug/Trace trigger Info register */
+register tinfo    : TInfo
+
+function isValidTselect(tselect_val : xlenbits) -> bool = {
+  let N_triggers = 3;// Number of triggers supported
+  (unsigned(tselect_val) >= 0) & (unsigned(tselect_val) < N_triggers);
+}  
+
+function legalize_tselect(o : TSelect, v : xlenbits) -> TSelect = {
+  let v = Mk_TSelect(v);
+  // Update Index field only if Trigger Module is enabled and Written value is legal
+  let o = [o with Index  = if (sys_enable_sdtrig() & isValidTselect(v[Index])) then v[Index] else o[Index]];
```

Comment:
I don't `legalize_tselect`  covers all the required legalisation.

Reply:
Since debug spec is not implemented in SAIL and it's still beyond scope, I don't think any other legalization is required.